File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(int n, int m) {
    vector<int> answer = {1, 0};
    int larger = n > m ? n : m;
    int smaller = n + m - larger;
    for(int i = 2; i <= larger; i)
    {
        if(larger % i == 0 && smaller % i == 0)
        {
            larger /= i;
            smaller /= i;
            answer[0] *= i;
            i = 2;
            continue;
        }
        i++;
    }
    answer[1] = larger * smaller * answer[0];
    return answer;
}

File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.cs
public class Solution {
    public int[] solution(int n, int m) {
        int[] answer = new int[2] {1, 1};
        if(n > m)
        {
            int temp = n;
            n = m;
            m = temp;
        }
        for(int i = 2; i <= n; i++)
        {
            if(m%i == 0 && n%i == 0)
                answer[0] = i;
        }
        int n2 = n;
        int m2 = m;
        while(n2 != m2)
        {
            if(n2 > m2) m2 += m;
            else if(n2 < m2) n2 += n;
        }
        answer[1] = n2;

        return answer;
    }
}